File: src/main/java/no/sanchezrolfsen/framework/selenium/elements/Checkbox.java
package no.sanchezrolfsen.framework.selenium.elements;

import no.sanchezrolfsen.framework.selenium.Browser;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Checkbox {
    private final By selector;

    public Checkbox(By by) {
        this.selector = by;
    }

    public void click() {
        getLabel().click();
    }

    public void waitAndClick() {
        getLabel().waitAndClick();
    }

    public String getText() {
        return getLabel().getText();
    }

    public boolean isDisplayed() {
        return getLabel().isDisplayed();
    }

    public boolean isChecked() {
        return getElement().isSelected();
    }

    public boolean isUnchecked() {
        return !isChecked();
    }

    public WebElement getElement() {
        return this.getDriver().findElement(this.selector);
    }

    public Label getLabel() {
        return new Label(getElement());
    }

    private WebDriver getDriver() {
        return Browser.vanillaDriver();
    }
}

File: src/main/kotlin/no/sanchezrolfsen/framework/selenium/elements/Checkbox.kt
package no.sanchezrolfsen.framework.selenium.elements

import no.sanchezrolfsen.framework.selenium.Browser
import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

class Checkbox(private val selector: By) {

    fun click() {
        getLabel().click()
    }

    fun waitAndClick() {
        getLabel().waitAndClick()
    }

    fun getText(): String = getLabel().getText()

    fun isDisplayed(): Boolean = getLabel().isDisplayed()

    fun isChecked(): Boolean = getElement().isSelected

    fun isUnchecked(): Boolean = !isChecked()

    fun getElement(): WebElement = getDriver().findElement(selector)

    fun getLabel(): Label = Label(getElement())

    private fun getDriver(): WebDriver = Browser.vanillaDriver()
}
